packages/flutter/test/material/selection_area_test.dart

```diff
@@ -133,4 +133,56 @@ void main() {
     expect(content, isNotNull);
     expect(content!.plainText, 'How');
   });
+
+  testWidgets('stopping drag of end handle will show the toolbar', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/119314
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: Padding(
+            padding: const EdgeInsets.only(top: 64),
+            child: Column(
+              children: <Widget>[
+                const Text('How are you?'),
+                SelectionArea(
+                  focusNode: FocusNode(),
+                  child: const Text('Good, and you?'),
+                ),
+                const Text('Fine, thank you.'),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+    final RenderParagraph paragraph2 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Good, and you?'), matching: find.byType(RichText)));
+    final TestGesture gesture = await tester.startGesture(textOffsetToPosition(paragraph2, 7)); // at the 'a'
+    addTearDown(gesture.removePointer);
+    await tester.pump(const Duration(milliseconds: 500));
+    await gesture.up();
+    await tester.pump(const Duration(milliseconds: 500));
+    final List<TextBox> boxes = paragraph2.getBoxesForSelection(paragraph2.selections[0]);
+    expect(boxes.length, 1);
+    // There is a selection now.
+    // We check the presence of the copy button to make sure the selection toolbar
+    // is showing.
+    expect(find.text('Copy'), findsOneWidget);
+
+    // This is the position of the selection handle displayed at the end
+    final Offset handlePos = paragraph2.localToGlobal(boxes[0].toRect().bottomRight);
+    await gesture.down(handlePos);
+    await gesture.moveTo(textOffsetToPosition(paragraph2, 11) + Offset(0, paragraph2.size.height / 2));
+    await tester.pump(const Duration(milliseconds: 500));
+    await tester.pump(const Duration(milliseconds: 500));
+
+    await gesture.up();
+    await tester.pump(const Duration(milliseconds: 500));
+    await tester.pump(const Duration(milliseconds: 500));
```

Comment:
Can you replace these `pump`s with `pumpAndSettle` or, if possible, a `pump` with no Duration?  Ideally we shouldn't have any arbitrary timing things like this unless it's totally necessary.

Reply:
I was able to remove most of the `pump`s with duration, except for one location. I believe this one is necessary though, because it is also used in other tests.

For example, here:
https://github.com/flutter/flutter/blob/686266e4184a107b5e52ddc871cb5400d8caf9df/packages/flutter/test/material/selection_area_test.dart#L61

Reply:
Sounds good, it's fine like this then 👍

Reply:
I think this particular use of `pump` is so the down event is held and detected as a long press.